nevergrad/parametrization/_datalayers.py

```diff
@@ -295,3 +297,54 @@ def _layered_set_value(self, value: tp.Any) -> tp.Any:
         coeff = discretization.weight_for_reset(self.arity)
         out[np.arange(value.size, dtype=int), value] = coeff
         super()._layered_set_value(out)
+
+
+class AngleOp(Operation):
+    """Converts to and from angles from -pi to pi"""
+
+    def _layered_get_value(self) -> tp.Any:
+        x = super()._layered_get_value()
+        if not x.shape[0] == 4:
+            raise ValueError("First dimension should be 4, got {x.shape}")
+        return np.angle(x[0, ...] - x[2, ...] + (x[1, ...] - x[3, ...]) * 1j)
```

Comment:
@teytaud 
Should `x` be constrained to be between 0 and 1? or should it be left "free"?
For constraining, there would be several options:
- reset the representation from the value of the angle
- add actual bounds on x so that it is either mapped to 0,1 (using arctan) or bounded using clipping/bouncing for instance